perma_web/api/resources.py

```diff
@@ -508,25 +520,43 @@ def obj_create(self, bundle, **kwargs):
         return bundle
 
     def obj_update(self, bundle, skip_errors=False, **kwargs):
-        is_private = bundle.obj.is_private
-        bundle = super(LinkResource, self).obj_update(bundle, skip_errors, **kwargs)
+        uploaded_file = bundle.data.get('file')
+        if uploaded_file and bundle.request.method == 'PATCH':
+            if kwargs['request']:
+                del kwargs['request']
 
```

Comment:
request ends up in kwargs because of the hacky patch_detail up above. If it exists in kwargs, delete! Otherwise everything will fail.